autopush/tests/test_integration.py

```diff
@@ -991,6 +997,155 @@ def test_message_update_unackd_delivered_message(self):
         eq_(result["data"], urlsafe_b64encode(data2))
         yield self.shut_down(client)
 
+    @inlineCallbacks
+    def test_webpush_monthly_rotation(self):
+        from autopush.db import make_rotating_tablename
+        client = yield self.quick_register(use_webpush=True)
+        yield client.disconnect()
+
+        # Move the client back one month to the past
+        last_month = make_rotating_tablename(
+            prefix=self._settings._message_prefix, delta=-1)
+        lm_message = self._settings.message_tables[last_month]
+        yield deferToThread(
+            self._settings.router.update_message_month,
+            client.uaid,
+            last_month
+        )
+
+        # Verify the move
+        c = yield deferToThread(self._settings.router.get_uaid, client.uaid)
+        eq_(c["current_month"], last_month)
+
+        # Move the clients channels back one month
+        exists, chans = yield deferToThread(
+            self._settings.message.all_channels, client.uaid
+        )
+        eq_(exists, True)
```

Comment:
Ah, we're checking the `exists` value here. Probably want to replace `exists` with `_` in `test_save_channels` and `test_all_channels_no_uaid`